src/main/java/org/jabref/gui/maintable/FieldColumn.java

```diff
@@ -18,44 +15,40 @@
 import org.jabref.model.entry.field.InternalField;
 import org.jabref.model.entry.field.OrFields;
 
-public class NormalTableColumn extends MainTableColumn<String> {
+/**
+ * A column that displays the text-value of the field
+ */
+public class FieldColumn extends MainTableColumn<String> {
 
     private final OrFields bibtexFields;
 
-    private final boolean isIconColumn;
-
-    private final Optional<JabRefIcon> iconLabel;
-
     private final Optional<BibDatabase> database;
 
     private final LayoutFormatter toUnicode = new LatexToUnicodeFormatter();
-    private final String columnName;
 
-    public NormalTableColumn(String columnName, OrFields bibtexFields, BibDatabase database) {
-        super(columnName);
-        this.columnName = columnName;
+    public FieldColumn(MainTableColumnModel model, OrFields bibtexFields, BibDatabase database) {
+        super(model);
         this.bibtexFields = bibtexFields;
-        this.isIconColumn = false;
-        this.iconLabel = Optional.empty();
         this.database = Optional.of(database);
+
+        setText(getDisplayName());
+        setCellValueFactory(param -> getColumnValue(param.getValue()));
     }
 
     /**
      * Get the table column name to be displayed in the UI
      *
      * @return name to be displayed. null if field is empty.
      */
-    public String getDisplayName() {
-        return bibtexFields.getDisplayName();
-    }
-
     @Override
+    public String getDisplayName() { return bibtexFields.getDisplayName(); }
+
     public ObservableValue<String> getColumnValue(BibEntryTableViewModel entry) {
         if (bibtexFields.isEmpty()) {
             return null;
         }
 
-        ObjectBinding<Field>[] dependencies = bibtexFields.stream().map(entry::getField).toArray(ObjectBinding[]::new);
+        ObjectBinding[] dependencies = bibtexFields.stream().map(entry::getField).toArray(ObjectBinding[]::new);
```

Comment:
JabRef refactored this line by itself and I did not see it. Should I undo it?

Reply:
Looks fine to me. Seems like the compiler is fine with infering the type

Reply:
oops, i meant intellij did this. 😅